modules/js/perf/perf_imgproc/perf_cvtcolor.js

```diff
@@ -0,0 +1,534 @@
+
+    "COLOR_RGBA2BGR", "COLOR_RGBA2GRAY",
+    "CX_RGBA2HLS", "CX_RGBA2HLS_FULL", "CX_RGBA2HSV", "CX_RGBA2HSV_FULL",
+    "CX_RGBA2Lab", "CX_RGBA2Luv", "CX_RGBA2XYZ",
+    "CX_RGBA2YCrCb", "CX_RGBA2YUV",
+
+    "COLOR_XYZ2BGR", "COLOR_XYZ2RGB", "CX_XYZ2BGRA", "CX_XYZ2RGBA",
+
+    "COLOR_YCrCb2BGR", "COLOR_YCrCb2RGB", "CX_YCrCb2BGRA", "CX_YCrCb2RGBA",
+    "COLOR_YUV2BGR", "COLOR_YUV2RGB", "CX_YUV2BGRA", "CX_YUV2RGBA"
+  ];
+  const CvtMode32FSize = [cvSize.szODD, cvSize.szVGA, cvSize.sz1080p];
+  const combiCvtMode32F = help_func.combine(CvtMode32FSize, CvtMode32F);
+
+  const CvtModeBayer = [
+    "COLOR_BayerBG2BGR", "COLOR_BayerBG2BGRA", "COLOR_BayerBG2BGR_VNG", "COLOR_BayerBG2GRAY",
+    "COLOR_BayerGB2BGR", "COLOR_BayerGB2BGRA", "COLOR_BayerGB2BGR_VNG", "COLOR_BayerGB2GRAY",
+    "COLOR_BayerGR2BGR", "COLOR_BayerGR2BGRA", "COLOR_BayerGR2BGR_VNG", "COLOR_BayerGR2GRAY",
+    "COLOR_BayerRG2BGR", "COLOR_BayerRG2BGRA", "COLOR_BayerRG2BGR_VNG", "COLOR_BayerRG2GRAY"
+  ];
+  const CvtModeBayerSize = [cvSize.szODD, cvSize.szVGA];
+  const combiCvtModeBayer = help_func.combine(CvtModeBayerSize, CvtModeBayer);
+
+
+  const CvtMode2 = [
+    "COLOR_YUV2BGR_NV12", "COLOR_YUV2BGRA_NV12", "COLOR_YUV2RGB_NV12", "COLOR_YUV2RGBA_NV12", "COLOR_YUV2BGR_NV21", "COLOR_YUV2BGRA_NV21", "COLOR_YUV2RGB_NV21", "COLOR_YUV2RGBA_NV21",
+    "COLOR_YUV2BGR_YV12", "COLOR_YUV2BGRA_YV12", "COLOR_YUV2RGB_YV12", "COLOR_YUV2RGBA_YV12", "COLOR_YUV2BGR_IYUV", "COLOR_YUV2BGRA_IYUV", "COLOR_YUV2RGB_IYUV", "COLOR_YUV2RGBA_IYUV",
+    "COLOR_YUV2GRAY_420", "COLOR_YUV2RGB_UYVY", "COLOR_YUV2BGR_UYVY", "COLOR_YUV2RGBA_UYVY", "COLOR_YUV2BGRA_UYVY", "COLOR_YUV2RGB_YUY2", "COLOR_YUV2BGR_YUY2", "COLOR_YUV2RGB_YVYU",
+    "COLOR_YUV2BGR_YVYU", "COLOR_YUV2RGBA_YUY2", "COLOR_YUV2BGRA_YUY2", "COLOR_YUV2RGBA_YVYU", "COLOR_YUV2BGRA_YVYU"
+  ];
+  const CvtMode2Size = [cvSize.szVGA, cvSize.sz1080p, cvSize.sz130x60];
+  const combiCvtMode2 = help_func.combine(CvtMode2Size, CvtMode2);
+
+  const CvtMode3 = [ 
+    "COLOR_RGB2YUV_IYUV", "COLOR_BGR2YUV_IYUV", "COLOR_RGBA2YUV_IYUV", "COLOR_BGRA2YUV_IYUV",
+    "COLOR_RGB2YUV_YV12", "COLOR_BGR2YUV_YV12", "COLOR_RGBA2YUV_YV12", "COLOR_BGRA2YUV_YV12"
+  ];
+  const CvtMode3Size = [cvSize.szVGA, cvSize.sz720p, cvSize.sz1080p, cvSize.sz130x60];
+  const combiCvtMode3 = help_func.combine(CvtMode3Size, CvtMode3);
+
+  const EdgeAwareBayerMode = [
+    "COLOR_BayerBG2BGR_EA", "COLOR_BayerGB2BGR_EA", "COLOR_BayerRG2BGR_EA", "COLOR_BayerGR2BGR_EA"
+  ];
+  const EdgeAwareBayerModeSize = [cvSize.szVGA, cvSize.sz720p, cvSize.sz1080p, cvSize.sz130x60];
+  const combiEdgeAwareBayer = help_func.combine(EdgeAwareBayerModeSize, EdgeAwareBayerMode);
+  
+  function getConversionInfo(cvtMode)
+  {
+    let mode;
+    switch(cvtMode)
+    {
+      case "COLOR_BayerBG2GRAY": case "COLOR_BayerGB2GRAY":
+      case "COLOR_BayerGR2GRAY": case "COLOR_BayerRG2GRAY":
+      case "COLOR_YUV2GRAY_420":
+          return [1, 1];
```

Comment:
why not use cv constants directly, e.g. `cv.CV_8UC`?

Reply:
Because I need to print these params with console log. If I use cv constants directly, I will need another mapping between  cv constants and related string for printing.

Reply:
Could you please point out where you need to print these params in log.

Reply:
Please see here https://github.com/huningxin/opencv/blob/838662834ca5ab73af12e1072d5a042021876dda/modules/js/perf/perf_imgproc/perf_cvtcolor.js#L516
https://github.com/huningxin/opencv/blob/838662834ca5ab73af12e1072d5a042021876dda/modules/js/perf/perf_imgproc/perf_resize.js#L198

And the log is just like: `params: (CV_8UC4,1280x720,640x480)`

Reply:
Thanks for the pointers. However I am still not convinced by the usage. The mapping from cv constants to string representation seems straightforward to me.

Reply:
Here is for the data type extension in the future. Now we just support `8U`(Uint8), and may support `S16` and `F32` later. And I did some changes in the latest commit https://github.com/huningxin/opencv/pull/262/commits/836c62b05f18c7a3e1b13c23693cd43f50078b0e for you to see the code more clearly.

Reply:
Please add a comment to explain the output, for example the "the channel number of source and destination mat". With that, I am fine to merge.

Reply:
OK. The returned array means the channel number of source and destination mat. For example, [4, 1] means that the source mat type is "8UC4" and destination mat type is "8UC1".